tests/unit/preprocessor/_time/test_time.py

```diff
@@ -55,6 +51,12 @@ def _create_sample_cube(calendar='gregorian'):
     return cube
 
 
+def assert_equal_and_same_dtype(result, expected):
+    """"Check data is identical to the expected one."""
```

Comment:
all the `_time` functions apart from `climate_statistics` **should not** alter the `dtype` but if they do, there is no explicit check for that inside the respective function, so the test will fail with a very obscure assertion error, it's a good idea to plop a print statement here a la "Checking for dtype consistency between original cube data and preprocessor result" and then the developer will know what hit them